frontend/src/container/JoinWorkspace.jsx

```diff
@@ -0,0 +1,183 @@
+import {
+  PAGE
+} from '../util/helper.js'
+import { getWorkspaceSubscriptions, postUserWorkspace, putUserWorkspaceSubscription } from '../action-creator.async.js'
+
+require('../css/JoinWorkspace.styl')
+
+export class JoinWorkspace extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = { filter: '' }
+
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+  }
+
+  async componentDidMount () {
+    this.handleAllAppChangeLanguage()
+    await this.loadWorkspaceSubscriptions()
+  }
+
+  async loadWorkspaceSubscriptions () {
+    const { props } = this
+
+    const fetchSubscriptionList = await props.dispatch(getWorkspaceSubscriptions(props.user.userId))
+
+    if (fetchSubscriptionList.status !== 200) {
+      props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+      return
+    }
+    props.dispatch(setWorkspaceSubscriptionList(fetchSubscriptionList.json))
+  }
+
+  handleAllAppChangeLanguage () {
+    const { props } = this
+    this.buildBreadcrumbs()
+    props.dispatch(setHeadTitle(props.t('Join a space')))
+  }
+
+  buildBreadcrumbs () {
+    const { props } = this
+
+    props.dispatch(setBreadcrumbs([{
+      link: <Link to={PAGE.HOME}><i className='fa fa-home' />{props.t('Home')}</Link>,
+      type: BREADCRUMBS_TYPE.CORE
+    }, {
+      link: <Link to={PAGE.JOIN_WORKSPACE}>{props.t('Join a space')}</Link>,
+      type: BREADCRUMBS_TYPE.CORE
+    }]))
+  }
+
+  async joinWorkspace (workspaceId) {
+    const { props } = this
+    await props.dispatch(postUserWorkspace(workspaceId, props.user.userId))
+    props.history.push(PAGE.WORKSPACE.DASHBOARD(workspaceId))
+  }
+
+  createRequestComponent (workspace) {
+    const { props } = this
+
+    const subscription = props.workspaceSubscriptionList.find(s => s.workspace.workspace_id === workspace.id)
+
+    if (subscription !== undefined && subscription.state !== SUBSCRIPTION_TYPE.rejected.slug) {
+      let text = 'Unknown request state'
+      let icon = 'question'
+      switch (subscription.state) {
+        case SUBSCRIPTION_TYPE.pending.slug:
+          text = props.t('Request sent')
+          icon = SUBSCRIPTION_TYPE.pending.faIcon
+          break
+      }
+      return <div><i class={`fa fa-${icon}`} /> {text}</div>
+    }
+
+    switch (workspace.accessType) {
+      case SPACE_TYPE.onRequest.slug:
+        return (
+          <IconButton
+            icon='share'
+            text={props.t('Request access')}
+            onClick={() => props.dispatch(putUserWorkspaceSubscription(workspace.id, props.user.userId))}
+          />)
+      case SPACE_TYPE.open.slug:
+        return (
+          <IconButton
+            icon='sign-in'
+            text={props.t('Join the space')}
+            onClick={() => this.joinWorkspace(workspace.id)}
+          />)
+      default:
+        return <span>Unknown space access type</span>
```

Comment:
```suggestion
        return <span>Unknown space access type</span>
```
I know this is a "should never happen" kind of message, but should this be translated anyway?